pyiceberg/catalog/glue.py

```diff
@@ -84,19 +110,105 @@ def _construct_parameters(
     return new_parameters
 
 
+def _type_to_glue_type_string(input_type: IcebergType) -> str:
```

Comment:
I think we might utilize the existing schema visitor for this task:
https://github.com/apache/iceberg-python/blob/4616d036440bf4cb3733e8d091220587cf290b75/pyiceberg/schema.py#L345-L348
https://github.com/apache/iceberg-python/blob/4616d036440bf4cb3733e8d091220587cf290b75/pyiceberg/schema.py#L784-L797

We could implement a `_TypeToGlueTypeString` implementing the `SchemaVisitor[str]` like this
```python
class _TypeToGlueTypeString(SchemaVisitor[str]):
    def schema(self, schema: Schema, struct_result: str) -> str:
        return struct_result

    def struct(self, struct: StructType, field_results: List[str]) -> str:
        name_to_type = ",".join(field_results)
        return f"struct<{name_to_type}>"

    def field(self, field: NestedField, field_result: str) -> str:
        return f"{field.name}:{field_result}"

    def list(self, list_type: ListType, element_result: str) -> str:
        return f"array<{element_result}>"

    def map(self, map_type: MapType, key_result: str, value_result: str) -> str:
        return f"map<{key_result},{value_result}>"

    def primitive(self, primitive: IcebergType) -> str:
        if isinstance(primitive, BooleanType):
            return "boolean"
        elif isinstance(primitive, IntegerType):
            return "int"
        elif isinstance(primitive, LongType):
            return "bigint"
        ...(more types)

        raise ValueError(f"Unknown Type {primitive}")
```

This can also let us benefit from the `@singledispatch` decorator, which make the visitor faster in practice. Does this idea sound reasonable to you?

Reply:
Yes, definitely, I didn't catch this in the other code. Will do.

Reply:
Done